### PR TITLE
Adds support for custom TTL of repaired docs and ViewQueryPageSize defaults

### LINKED ISSUE #2892: Repair tool efficiency improvements
The repair tool in https://github.com/couchbase/sync_gateway/issues/2857 needs efficiency improvements for iterating large datasets.

### PR DESCRIPTION
**Pull request by ajres**
fixes #2892

### CODE REVIEW

**Comment by adamcfraser** [comment]
Can you recreate this PR against 1.3.1.4?

**Comment by ajres** [comment]
@adamcfraser done, it was coded off release/1.3.1.4 branch, but I ended up on a detached HEAD when I pushed to GitHub.

File: db/repair_bucket.go [hunk 1]
```diff
@@ -6,19 +6,23 @@ import (
...
 const (
 	RepairRevTreeCycles = RepairJobType("RepairRevTreeCycles")
 )
 
 // Params suitable for external (eg, HTTP) invocations to describe a RepairBucket operation
 type RepairBucketParams struct {
 	DryRun            bool              `json:"dry_run"`
-	ViewQueryPageSize int               `json:"view_query_page_size"`
+	ViewQueryPageSize *int              `json:"view_query_page_size"`
+	RepairedFileTTL   *int              `json:"repaired_file_ttl_seconds"`
```
Comment (adamcfraser) [inline db/repair_bucket.go]: Can we rename this to "repair_output_ttl_seconds"?  As written, it sounds like the repaired file itself gets a ttl, which is a bit confusing.

**Review by adamcfraser — CHANGES_REQUESTED** [review]

File: db/repair_bucket.go (hunk 1 above)
Comment (ajres) [reply db/repair_bucket.go]: Modified parameter name is pushed up

**Pull request merged**